### ISSUE TITLE
Include all symmetries during training

### ISSUE THREAD

**Issue by MaMiFreak** [Rochester-NRT/RocAlphaGo#188]
https://github.com/MaMiFreak/RocAlphaGo/pull/5/files

i made an implementation that includes all symmetries in the shuffle file
instead of noop/diag2 i used 0/8 or the shuffle file will become to big

my questions
- do we want the option to remove certain symmetries? or would true/false suffice to include all or none
- i added a max-validation argument that is able to override the --train-val-test fraction setting, agree?
- test set is not used, do we want to do anything with it?
- any concerns on the code used or improvements?

**Comment by wrongu** [comment]
* I like the idea of adding an `'all'` or `'none'` option (`'none'` would, of course, default to `noop`)

        ... --symmetries noop,rot90,rot180,rot270
        ... --symmetries all
        ... --symmetries none

* agreed. Makes sense when there is a separate shuffle file for each.
* I'd like to leave the test set in for future use. The idea behind a separate test set is that it serves as the validation for the validation set. In other words, you do model selection with the validation set (e.g. early stopping) but should ultimately evaluate test error on the test set
* I'll add comments on the code itself if I see anything

**Comment by MaMiFreak** [comment]
@wrongu thanks for the feedback! 

i finished the implementation, still have to change some comments.

biggest issue remaining is how to validate the same .hdf5 feature file and symmetries are used and what to do when they are different...

- create test/train/validation shuffle files with all symmetries and filter them at runtime
( advantage would be that validation/test/training stays separated even when restarting with different settings, downside bigger shuffle file size.. and additional validation has to be done when training)
- generate new shuffle files when using different settings..

also what to do when a different fraction is used.

Now, writing this down i'd suggest to create test/train/validation shuffle files with all symmetries and create a new one when using a different fraction setting. 

Also implemented test set file but i set the default fragment to 0. i will add a TODO for future when we will implement something for the test set

**Comment by MaMiFreak** [comment]
is there a way to uniquely identify a hdf5 file? i like to store this in the metadata file to make sure we are starting again with the same file.
if not a new shuffle file has to be created. 
is a function like that available, can't find anything like it in the docs.. could use `dataset["states"]` size alternatively.

**Comment by MaMiFreak** [comment]
downside of filtering unwanted symmetries during training is that we could get a 
`warning mentioning accuracy from keras`

**Comment by MaMiFreak** [comment]
https://github.com/MaMiFreak/RocAlphaGo/tree/MaMiFreak-symmetries
fixed #148 and #187
- restarting with sgd and decay does use wrong lr ( will be fixed with keras1.1.3 but i could implement it already)
- do you like the prompts when resume learning with different settings? i think for new users it is helpful 
- could somebody verify lr decay with theano backend?
- could somebody verify correct weight file names when restarting with theano backend
- any more requests?
- any errors?

**Comment by MaMiFreak** [comment]
Two features i am in doubt about
- backup: save model every x hours to recover from a crash
- separate resume learning start where no additional arguments are accepted and all are loaded from metadata

Question:
I would like to add a extended log option, tensorflow has a nice function that does log everything and is able to show it in tensorboard, does Theano have such an option?
